wagtail/admin/views/generic/models.py

```diff
@@ -241,19 +248,29 @@ def search_queryset(self, queryset):
         }
         return queryset.filter(**filters)
 
-    def _get_title_column(self, column_class=TitleColumn):
-        def title_accessor(obj):
+    def _get_field_accessor(self, field_name):
+        def accessor(obj):
             draftstate_enabled = self.model and issubclass(self.model, DraftStateMixin)
 
-            if draftstate_enabled and obj.latest_revision:
+            # For models with DraftStateMixin, use latest revision's object_str for __str__
+            if field_name == "__str__" and draftstate_enabled and obj.latest_revision:
                 return obj.latest_revision.object_str
-            return str(obj)
 
-        return column_class(
-            "name",
-            label=gettext_lazy("Name"),
-            accessor=title_accessor,
-            get_url=self.get_edit_url,
+            *_, value = lookup_field(field_name, obj)
+            return value
+
+        return accessor
+
+    def _get_title_column(self, field_name, column_class=TitleColumn):
+        if not self.model:
+            return column_class(
+                "name",
+                label=gettext_lazy("Name"),
+                accessor=self._get_field_accessor(field_name),
```

Comment:
`Column.accessor` already has [some pretty fancy logic for interpreting accessor strings](https://github.com/wagtail/wagtail/blob/d1908ee07f3959abfdd1d317ff2fa3598a27579e/wagtail/admin/ui/tables.py#L90-L93), including dict key lookups and dotted paths - are we actually gaining anything from passing this through `_get_field_accessor` and `lookup_field`, rather than just passing `accessor=field_name`? It seems like we're going to end up with two different conventions for what's valid as an accessor name, depending on whether we go through `Column(accessor=...)` or `list_display`.

Reply:
Good point! I've removed `_get_field_accessor` and moved the specific `DraftStateMixin` check to the snippets listing view, which I will extract into the view mixins later.